### PR TITLE
[TACHYON-442] Apply PluginManagement to cleanup pom.xml in each module

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-442

Further cleanup the pom files, focusing on maven-surefire-plugin and maven-compiler-plugin.

### CODE REVIEW

File: clients/unshaded/pom.xml [hunk 1]
```diff
@@ -5,7 +5,6 @@
     <artifactId>tachyon-clients</artifactId>
     <version>0.7.0-SNAPSHOT</version>
   </parent>
-  <groupId>org.tachyonproject</groupId>
```
Comment (haoyuan) [inline clients/unshaded/pom.xml]: why remove here?

Comment (apc999) [inline clients/unshaded/pom.xml]: The groupdId of a child pom is by default inherited from its parent pom. Before this PR, in our code some poms have defined the groupId (which is unnecessary anyway), while some other poms haven't. So in this PR, I just cleanup the poms by removing this redundant definitions to keep pom files consistent.

File: pom.xml [hunk 2]
```diff
@@ -274,37 +274,49 @@
   </dependencyManagement>
 
   <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
```
Comment (calvinjia) [inline pom.xml]: Will this configuration apply to all executions?

File: pom.xml [hunk 3]
```diff
@@ -274,37 +274,49 @@
...
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.2</version>
+          <configuration>
+            <source>${java.version}</source>
+            <target>${java.version}</target>
+            <encoding>UTF-8</encoding>
+            <maxmem>1024m</maxmem>
```
Comment (calvinjia) [inline pom.xml]: Are these four values necessary/good to have? Limiting `maxmem` may lead to problems?

**Comment by calvinjia** [comment]
+1 LGTM, much cleaner than before.

**Comment by haoyuan** [comment]
Thanks!

**Pull request merged**